.github/workflows/tests.yml

```diff
@@ -256,21 +260,42 @@ jobs:
         run: find /usr/local/faasm/object | grep aot | xargs rm -f
       # --- Codegen build without sanitisers
       - name: "Build codegen - CMake"
-        run: inv -r faasmcli/faasmcli dev.cmake --build Debug --sanitiser None
+        run: |
+          source venv/bin/activate
+          inv -r faasmcli/faasmcli dev.cmake --build Debug --sanitiser None
+        shell: bash
       - name: "Build codegen - Codegen Func"
-        run: inv -r faasmcli/faasmcli dev.cc codegen_func
+        run: |
+          source venv/bin/activate
+          inv -r faasmcli/faasmcli dev.cc codegen_func
+        shell: bash
       - name: "Build codegen - Codegen Shared Obj"
-        run: inv -r faasmcli/faasmcli dev.cc codegen_shared_obj
+        run: |
+          source venv/bin/activate
```

Comment:
Can we avoid this repetition with a [custom shell](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#custom-shell) or by setting up an alias for `inv` at the start?

Reply:
I set a default shell (that activates the virtual env. before running the command) only for the job that runs the tests.

Note that we can't set the shell as a default for all the workflow as not all jobs run in the same container (thus don't have the venv set, neither they need it).